com.unity.template.renderstreaming-hd/Assets/Scripts/Signaling.cs

```diff
@@ -1,272 +1,544 @@
+
+            DescData data = new DescData();
+            data.connectionId = connectionId;
+            data.sdp = answer.sdp;
+            data.type = "answer";
+
+            if (_uri.Scheme == "http" || _uri.Scheme == "https") {
+                HTTPPost("signaling/answer", data);
+            } else {
+
+                RoutedMessage<DescData> routedMessage = new RoutedMessage<DescData>();
+                routedMessage.to = connectionId;
+                routedMessage.message = data;
+
+                WSSend(routedMessage);
+            }
+
+        }
+
+
+        //------------------------------------------------------------------------------------------
+        //WebSocket Version
+        //------------------------------------------------------------------------------------------
+
+
+        public void WSManage() {
+
+            while (_running) {
+                WSCreate();
+
+                _wsCloseEvent.WaitOne();
+
+
+
+                Thread.Sleep((int)(_timeout * 1000));
+            }
+
+            Debug.Log("Signaling: WS managing thread ended");
+        }
+
+        private void WSCreate() {
+
+
+            const SslProtocols sslProtocolHack = (SslProtocols)(SslProtocolsHack.Tls12 | SslProtocolsHack.Tls11 | SslProtocolsHack.Tls);
+
+            _webSocket = new WebSocketSharp.WebSocket(_uri.ToString());
+            if (_uri.Scheme == "wss"){
+                _webSocket.SslConfiguration.EnabledSslProtocols = sslProtocolHack;
+            }
+
+            _webSocket.OnOpen += WSConnected;
+            _webSocket.OnMessage += WSProcessMessage;
+            _webSocket.OnError += WSError;
+            _webSocket.OnClose += WSClosed;
+
+            Monitor.Enter(_webSocket);
+
+            Debug.Log($"Signaling: Connecting WS {_uri.ToString()}");
+            _webSocket.ConnectAsync();
+
+        }
+
+        private void WSProcessMessage(object sender, MessageEventArgs e){
+
+            var content = Encoding.UTF8.GetString(e.RawData);
+            Debug.Log($"Signaling: Receiving message: {content}");
+
+            try {
+
+                var routedMessage = JsonUtility.FromJson<RoutedMessage<SignalingMessage>>(content);
+
+                SignalingMessage msg;
+                if (!string.IsNullOrEmpty(routedMessage.from)) {
+                    msg = routedMessage.message;
+                } else {
+                    msg = JsonUtility.FromJson<SignalingMessage>(content);
+                }
+
+                if (!string.IsNullOrEmpty(msg.type)){
+
+                    if (msg.type == "signIn") {
+
+                        if (msg.status == "SUCCESS"){
+
+                            this._connectionId = msg.connectionId;
+                            this._sessionId = msg.peerId;
+                            Debug.Log("Signaling: Slot signed in.");
+
+                            this.WSSend("{\"type\":\"furioos\",\"task\":\"enableStreaming\",\"streamTypes\":\"WebRTC\",\"controlType\":\"RenderStreaming\"}");

```

Comment:
Need to describe in the comment what this is

Reply:
ok
To be later, I will separate Furioos WS and Sample WS.